Userland/Libraries/LibGfx/ImageFormats/BooleanDecoder.cpp

```diff
@@ -82,23 +85,26 @@ ErrorOr<bool> BooleanDecoder::read_bool(u8 probability)
     m_value <<= bits_to_shift_into_range;
     m_value_bits_left -= bits_to_shift_into_range;
 
-    TRY(fill_reservoir());
+    fill_reservoir();
 
     return return_bool;
 }
 
 // 9.2.4 Parsing process for read_literal
-ErrorOr<u8> BooleanDecoder::read_literal(u8 bits)
+u8 BooleanDecoder::read_literal(u8 bits)
 {
     u8 return_value = 0;
     for (size_t i = 0; i < bits; i++) {
-        return_value = (2 * return_value) + TRY(read_bool(128));
+        return_value = (2 * return_value) + read_bool(128);
     }
     return return_value;
 }
 
 ErrorOr<void> BooleanDecoder::finish_decode()
 {
+    if (m_overread)
+        return Error::from_string_literal("Range decoder was read past the end of its data");
```

Comment:
I guess that's why you did the change in the first commit. I'd add a separate `did_read_too_many_bits()` or similar thingy for this, probably (and call it fro the `finish_decode_vp9()` function, I suppose).